api/src/main/java/au/org/aodn/nrmn/restapi/service/CorrectionService.java

```diff
@@ -0,0 +1,88 @@
+package au.org.aodn.nrmn.restapi.service;
+
+import au.org.aodn.nrmn.restapi.model.db.*;
+import au.org.aodn.nrmn.restapi.repository.ObservationRepository;
+import au.org.aodn.nrmn.restapi.repository.SurveyMethodRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Example;
+import org.springframework.stereotype.Service;
+
+import java.sql.Date;
+import java.sql.Time;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Service
+public class CorrectionService {
+    @Autowired
+    private ObservationRepository observationRepository;
+    @Autowired
+    private SurveyMethodRepository surveyMethodRepository;
+
+    public Stream<StagedRow> convertSurveyToStagedRows(Survey survey) {
+        List<SurveyMethod> surveyMethods = surveyMethodRepository.findAll(Example.of(
+                SurveyMethod.builder().survey(survey).build()));
+        return surveyMethods.stream().map(this::convertSurveyMethodToStagedRow);
+    }
+
+    private StagedRow convertSurveyMethodToStagedRow(SurveyMethod surveyMethod) {
+        Survey survey = surveyMethod.getSurvey();
+        StagedRow.StagedRowBuilder builder = StagedRow.builder()
+                .depth(getStagedDepth(survey.getDepth(), survey.getSurveyNum()))
+                .siteNo(survey.getSite().getSiteCode())
```

Comment:
sounds like a siteCode

Reply:
"siteNo" is used for StagedRow since it represents an spreadsheet row, and the header title in the excel spreadsheets is "siteNo". In the DB we use the more descriptive/accurate name of siteCode.